lib/ecto/adapters/mysql/connection.ex

```diff
@@ -377,8 +374,8 @@ if Code.ensure_loaded?(Mariaex) do
       ["NOT (", expr(expr, sources, query), ?)]
     end
 
-    defp expr(%Ecto.SubQuery{query: query, fields: fields}, _sources, _query) do
-      query.select.fields |> put_in(fields) |> all()
+    defp expr(%Ecto.SubQuery{query: query}, _sources, _query) do
```

Comment:
If you want to remain backwards compatible, use this implementation for subqueries:

```
defp expr(%Ecto.SubQuery{query: query} = subquery, _sources, _query) do
  if Map.has_key?(subquery, :fields) do
    query.select.fields |> put_in(subquery.fields) |> all()
  else
    all(query)
  end
end
```